src/index.ts

```diff
@@ -14,24 +15,28 @@ import {
 import { JSFOptions, JSFCallback } from './utils';
 
 export interface MiddlewareOptions {
-  file: string;
+  file: string | OpenAPIV3.Document;
```

Comment:
The option name seems not right now. I mean it's not a `file` already.
Maybe we could add a new option `spec`, and leave `file` as it was and mark it as deprecated option for backward compatibility?

Reply:
inMemory removed and spec added. "spec" is optional and can be undefined. in this case, it will be populated by the value of "file" which is optional as well (if user want to provide spec only, file will not be required)
to prevent users from forgetting to add "file" or "spec", another condition for check for undefined added.